sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -873,7 +873,7 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       .innerJoin("institutionLocation.institution", "institution")
       .leftJoin("offering.assessedBy", "assessedBy")
       .leftJoin("offering.precedingOffering", "precedingOffering")
-      .innerJoin("offering.parentOffering", "parentOffering")
```

Comment:
If this is the fix, I am concerned if the bug is real. Because, the following ticket ensured that parentId must be always present. https://github.com/bcgov/SIMS/issues/2488

Reply:
I have put some effort here and found the root cause. From where the overlap validation method is called, we should not use the `payload.data.studystartDate`  and `payload.data.studyendDate` 

![image](https://github.com/user-attachments/assets/87c829cf-b6ff-4634-8dad-ffc7cb718ab9)

THese fields would get populated only when student does not select the offering or program from student application which leads to PIR. 

![image](https://github.com/user-attachments/assets/f3fefc76-9f94-4883-9ae1-eeab49a9a9a1)

Reply:
Yes, the code is adjusted to accommodate that scenario.